base/io.jl

```diff
@@ -131,9 +125,20 @@ read{T}(s::IO, t::Type{T}, d1::Integer, dims::Integer...) =
 
 read{T}(s::IO, ::Type{T}, dims::Dims) = read!(s, Array(T, dims))
 
+function read!(s::IO, a::Vector{UInt8})
+    for i in 1:length(a)
+        a[i] = read(s, UInt8)
+    end
+end
+
 function read!{T}(s::IO, a::Array{T})
-    for i in eachindex(a)
-        a[i] = read(s, T)
+    if isbits(T)
+        nb::Int = length(a) * sizeof(T)
+        read!(s, reinterpret(UInt8, a, (nb,)))
```

Comment:
also, this method was never used since it is shadowed by re-implementations on all the subtypes used in the testing https://codecov.io/github/JuliaLang/julia/base/io.jl?ref=e82ba0cea7a131b06716d81fbcbf0cfd5feb4f59#l-134